packages/repository-json-schema/src/build-schema.ts

```diff
@@ -16,10 +16,44 @@ import {JSONSchema6 as JSONSchema} from 'json-schema';
 import {JSON_SCHEMA_KEY, MODEL_TYPE_KEYS} from './keys';
 
 export interface JsonSchemaOptions {
+  /**
+   * Set this flag if you want the schema to define navigational properties
+   * for model relations.
+   */
   includeRelations?: boolean;
+
+  /**
+   * Set this flag to mark all model properties as optional. This is typically
+   * used to describe request body of PATCH endpoints.
+   */
+  partial?: boolean;
+
+  /**
+   * @private
+   */
   visited?: {[key: string]: JSONSchema};
 }
 
+/**
+ * @private
+ */
+export function buildModelCacheKey(
+  ctor: Function,
+  options: JsonSchemaOptions = {},
+): string {
+  // backwards compatibility
+  switch (Object.keys(options).length) {
+    case 0:
+      return MODEL_TYPE_KEYS.ModelOnly;
+    case 1:
+      if (options.includeRelations) return MODEL_TYPE_KEYS.ModelWithRelations;
```

Comment:
Don’t you think cyclomatic complexity increases with if inside switch? Just a thought.

Reply:
Good point, I am going to simplify this function to make it easier to understand and reason about. Thank you for pointing it out! ❤️